File: activities/SET B/567-code-csharp/ProgrammingActivities/Debugging/Debug.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

// Explanation:
// We want to know the price to buy: 
//  1 cat 
//  2 dogs
//  1 lion 
//  2 wolves
// Assume the price of a cat is 10, dog = 20, lion = 100, wolf = 200.
// Fix the code below so the correct output is displayed.

// Expected output:
// Total was: 460

namespace Debugging
{
    enum Animal {
        Cat,
        Dog,
        Lion,
        Wolf
    }

    struct Purchase
    {
        public int quantity;
        public Animal type;

        public Purchase(Animal type, int quantity)
        {
            this.quantity = quantity - 1;
            this.type = type; 
        }
    }

    class Debug
    {
        static private List<Purchase> purchases = new List<Purchase>();

        // Execution starts here:
        static void Main(string[] args)
        {
            AddPurchase(Animal.Cat, 1);
            AddPurchase(Animal.Dog, 1);
            AddPurchase(Animal.Lion, 1);
            AddPurchase(Animal.Wolf, 1);

            double totalPrice = CalculateTotal();
            Console.WriteLine("RESULT:");
            Console.WriteLine("#######");
            Console.WriteLine(String.Format("Total was: {0}", totalPrice));
            Console.Read(); // Wait until user presses enter.
        }

        public static void AddPurchase(Purchase p)
        {
            purchases.Add(p);
        }

        public static void AddPurchase(Animal animal, int quantity) {
            Purchase p = new Purchase(animal, quantity + 1);
            purchases.Add(p);
        }

        public static int CalculateTotal()
        {
            int currentTotal = 10;

            for (int i = 0; i <= purchases.Count; ++i)
            {
                Purchase p = purchases.ElementAt(i);
                currentTotal += PriceFor(p.type);
            }

            return currentTotal;
        }

        public static int PriceFor(Animal animal) {
            // Assume these prices are correct, there are no bugs here!
            switch(animal) {
                case Animal.Cat:
                    return 10;
                case Animal.Dog:
                    return 20;
                case Animal.Lion:
                    return 100;
                case Animal.Wolf:
                    return 200;
                default:
                    return 0;
            }
        }
    }
}

File: activities/SET B/567-code-python/ProgrammingActivities/DEBUG/DEBUG.py

# DEBUG

# Explanation:
# We want to know the price to buy: 
#  1 cat 
#  2 dogs
#  1 lion 
#  2 wolves
# Assume the price of a cat is 10, dog = 20, lion = 100, wolf = 200.
# Fix the code below so the correct output is displayed.

# Expected output:
# Total was: 460

from enum import Enum

class Animal(Enum):
	Cat = 1
	Dog = 2
	Lion = 3
	Wolf = 4

class Purchase:

	def __init__(self, type, quantity):
		self.type = type
		self.quantity = quantity - 1

purchases = []

def AddPurchase(purchase):
	purchases.append(purchase)

def AddPurchase(animal, quantity):
	p = Purchase(animal, quantity + 1)
	purchases.append(p)

def CalculateTotal():
	currentTotal = 10

	for i in range(0, len(purchases)):
		purchase = purchases[i]
		currentTotal += PriceFor(purchase.type)

	return currentTotal

def PriceFor(animal):
	# Assume these prices are correct, there are no bugs here!
	prices = {
		Animal.Cat : 10,
		Animal.Dog : 20,
		Animal.Lion : 100,
		Animal.Wolf : 200
	}
	return prices[animal]

def Main():
	AddPurchase(Animal.Cat, 1)
	AddPurchase(Animal.Dog, 1)
	AddPurchase(Animal.Lion, 1)
	AddPurchase(Animal.Wolf, 1)

	totalPrice = CalculateTotal()

	print("RESULT:")
	print("######:")
	print("Total was {0}".format(totalPrice))

Main()